assembl/scripts/check_antivirus.py

```diff
@@ -0,0 +1,44 @@
+import argparse
+
+import transaction
+from sqlalchemy.orm import undefer
+
+from assembl.lib.sqla import mark_changed
+from assembl.lib.antivirus import get_antivirus
+from assembl.scripts import boostrap_configuration
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("configuration", help="configuration file")
+    parser.add_argument("-d", "--discussion", default=None,
+                        help="slug of single discussion to check (or all discussions)")
+    parser.add_argument("-r", "--reset", action="store_true",
+                        help="Reset status of all files")
+    parser.add_argument("-a", "--antivirus",
+                        default="assembl.lib.antivirus.SophosAntiVirus",
+                        help="antivirus class")
+    args = parser.parse_args()
+    session = boostrap_configuration(args.configuration)
+    antivirus = get_antivirus(args.antivirus)
+    from assembl.models import File, Discussion
+    from assembl.models.attachment import AntiVirusStatus
+    files = session.query(File)
+    if args.discussion:
+        (discussion_id,) = session.query(Discussion.id).filter_by(slug=args.discussion).first()
+        files = files.filter_by(discussion_id=discussion_id)
+    if args.reset:
+        with transaction.manager:
+            files.update({
+                File.av_checked: AntiVirusStatus.unchecked.name})
+            mark_changed(session)
+    with transaction.manager:
+        file_ids = files.with_entities(File.id).all()
```

Comment:
You could skip the `with_entities` step and just go straight to `files.all()`, and renaming `file_ids` to `file_objects`.

Reply:
it may be intentional so that query does not get file contents (the total can be huge)